sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -604,6 +604,25 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("bucket join should work with SubqueryAlias plan") {
+    withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "0") {
+      withTable("t") {
+        withView("v") {
+          val df = (0 until 20).map(i => (i, i)).toDF("i", "j").as("df")
+          df.write.format("parquet").bucketBy(8, "i").saveAsTable("t")
+
+          sql("CREATE VIEW v AS SELECT * FROM t").collect()
+
+          val plan1 = sql("SELECT * FROM t a JOIN t b ON a.i = b.i").queryExecution.executedPlan
+          assert(plan1.collect { case exchange: ShuffleExchangeExec => exchange }.isEmpty)
+
+          val plan2 = sql("SELECT * FROM t a JOIN v b ON a.i = b.i").queryExecution.executedPlan
```

Comment:
What does this test means? This test can improve the test coverage?

Reply:
I will remove `plan1` which is a benign case.